smokeTest.js

```diff
@@ -0,0 +1,112 @@
+process.argv.forEach((val, index) => {
+    console.log(`${index}: ${val}`);
+});
+  
+const XETHRU_STATE = require('./SessionStateXethruEnum.js')
+const MOV_THRESHOLD = 17
+const IM21_DOOR_STATUS = require('./IM21DoorStatusEnum')
+const testLocation1Id = 'TestLocation1'
+const door_coreID = 'door_coreID'
+const radar_coreID = 'radar_coreID'
+const { helpers } = require('brave-alert-lib')
+
+function getRandomInt(min, max) {
+    min = Math.ceil(min)
+    max = Math.floor(max)
+    return Math.floor(Math.random() * (max - min) + min) //The maximum is exclusive and the minimum is inclusive
+}
+
+function getRandomArbitrary(min, max) {
+    return Math.random() * (max - min) + min
+}
+
+
+let destinationURL = process.argv[2];
+let recipientNumber = process.argv[3];
+let twilioNumber = process.argv[4];
+
+
+  
+const axios = require('axios').default;
+const sleep = (millis) => new Promise(resolve => setTimeout(resolve, millis));
+axios.defaults.headers.post['Content-Type'] = 'application/x-www-form-urlencoded';
+axios.defaults.baseURL = destinationURL;
+      
+smokeTest(recipientNumber, twilioNumber);
+  
+async function smokeTest(recipientNumber, twilioNumber){  
+    await setup(recipientNumber, twilioNumber, testLocation1Id, door_coreID, radar_coreID)
+    await silence(testLocation1Id)
+    await im21Door(door_coreID, IM21_DOOR_STATUS.CLOSED)
+    for(let i = 0; i<15; i++){
+        await movement(testLocation1Id, getRandomInt(MOV_THRESHOLD + 1,100), getRandomInt(MOV_THRESHOLD + 1,100))
+    }
+    await im21Door(door_coreID, IM21_DOOR_STATUS.OPEN)
+    for(let i = 0; i<15; i++){
+        await movement(testLocation1Id, getRandomInt(0,MOV_THRESHOLD), getRandomInt(0,MOV_THRESHOLD))
+    }
+    await im21Door(door_coreID, IM21_DOOR_STATUS.CLOSED)
+    for(let i = 0; i<15; i++){
+        await movement(testLocation1Id, getRandomInt(0,MOV_THRESHOLD), getRandomInt(0,MOV_THRESHOLD))
+    }
+    for(let i = 0; i<15; i++){
+        await movement(testLocation1Id, getRandomInt(MOV_THRESHOLD + 1,100), getRandomInt(MOV_THRESHOLD + 1,100))
+    }
+    for(let i = 0; i<85; i++){
+        await silence(testLocation1Id)
+    }
+
+}
+    
+async function setup(recipientNumber, twilioNumber){
+    try{
+        await axios.post('/smokeTest/setup', {
+            recipientNumber: recipientNumber,
+            twilioNumber: twilioNumber,
+            locationid: testLocation1Id
+        })
+    }catch(e){helpers.log(e)}
+}
+
+async function silence(locationid){
```

Comment:
Is this exactly the same as in serverTest.js? If so, maybe a testHelper.js file would be useful for these sorts of things?

Reply:
they are yeah, I like that idea and will refactor

Reply:
they aren't the same right now, but they are very similar. The differences are the use of chai in tests vs axios in the smoketests. These are parameterized to access a provided URL, the serverTest.js ones have a hardcoded reference to the `server` object.

Reply:
looking at it now it seems chai can be used just fine for SmokeTests, so I think I'll still do a helper file with that

Reply:
so I ended up not doing a testHelpers file. I actually tried but kep trunning into issues using chai for url testing with a remote server. It seems like it really only works when you have an actual app/server object (like from express) to pass to it. Which is the case in the unit tests, but not in the smoke tests

Reply:
could still see an argument for a testHelpers file, but it would only have the getRandomInt and getRandomArbitrary functions :/ :shrug:

Reply:
Would probably be a good idea, and would also make it clear that such a place does exist for when other temptations to copy-and-paste crop up ;-)
But it's not a critical change, for me.

Reply:
I can see both angles of this too 😅  More helpery type functions will probably crop up over time so I feel like it will eventually exist regardless of when that happens